src/hooks/useRequest.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    AxiosError,
+    AxiosInstance,
+    AxiosRequestConfig,
+    AxiosResponse,
+} from "axios";
+import { useState } from "react";
+
+export const useRequest = (
```

Comment:
The hook seems nice, I think it'll really help in some scenarios such as upload for example.

I have an idea on how to get around the issue of having to pass `axiosInstance` in every call.
We could use [useContext](https://reactjs.org/docs/hooks-reference.html#usecontext) for this.

But, for this we'll need to have aardvark instantiated in the root of the application and then passed to a `AardvarkProvider` (which is currently in [test-utils](https://github.com/bornfight/aardvark/pull/39/files#diff-135dada2521405c112a7034de12d7876), but I think that the provider should be exported and required for implementation)